src/lib/web3-provider.ts

```diff
@@ -1,13 +1,26 @@
+import { avalancheChainId, ChainId, ethChainId } from '@shapeshiftoss/caip'
 import { getConfig } from 'config'
 import Web3 from 'web3'
 
-let maybeWeb3Provider: InstanceType<typeof Web3.providers.HttpProvider> | null
+type HttpProvider = InstanceType<typeof Web3.providers.HttpProvider>
 
-export const getWeb3Provider = () => {
-  if (!maybeWeb3Provider) {
-    maybeWeb3Provider = new Web3.providers.HttpProvider(getConfig().REACT_APP_ETHEREUM_NODE_URL)
-    return maybeWeb3Provider!
+const web3ProviderMap: Map<ChainId, HttpProvider> = new Map()
+
+const httpProviderByChain = (chainId: ChainId): string => {
+  switch (chainId) {
+    case avalancheChainId:
+      return getConfig().REACT_APP_AVALANCHE_NODE_URL
+    case ethChainId:
+    default:
+      return getConfig().REACT_APP_ETHEREUM_NODE_URL
+  }
+}
+
+export const getWeb3Provider = (chainId: ChainId): HttpProvider => {
```

Comment:
That might be better suited now that this is programmatic
```suggestion
export const getWeb3ProviderByChainId = (chainId: ChainId): HttpProvider => {
```

Reply:
Addressed in https://github.com/shapeshift/web/pull/2164/commits/aa5aaf46b2f8dc0b625bff71fb1c1c6008b93cf1